sys/net/application_layer/coap/gcoap.c

```diff
@@ -96,29 +433,100 @@ static void *_event_loop(void *arg)
                 case GCOAP_MSG_TYPE_INTR:
                     /* next _listen() timeout will account for open requests */
                     break;
+#ifdef MODULE_GNRC_DTLS
+                case DTLS_MSG_CLIENT_START:
+                  dtls_context_clnt = (dtls_context_t *) msg_rcvd.content.ptr;
+#      if ENABLE_DEBUG == 1
+                  if (!dtls_context_clnt)
+                    DEBUG("Client context not passed!\n");
+                  else
+                    DEBUG("DTLS now listen to client side only!\n");
+#      endif
+                    remote_aux =  (dtls_remote_peer_t *) dtls_get_app_data(dtls_context_clnt);
+                    msg_reply(&msg_rcvd,&msg_rcvd);
+                break;
+
+                case DTLS_MSG_SERVER_STOP:
+                    // dtls_free_context(dtls_context_srvr);
+                    // dtls_context_srvr = NULL;
+                    // dtls_context_clnt = NULL;
+                    flag = DTLS_FLAGS_CLIENT_LISTEN_ONLY;
+                    DEBUG("DTLS Server temporary disabled!\n");
+                    msg_reply(&msg_rcvd,&msg_rcvd);
+                break;
+                // case DTLS_MSG_CLIENT_DATA:
+                //     flag = DTLS_FLAGS_CLIENT_NON_LISTEN;
+                // break;
+
+                case DTLS_MSG_SERVER_RESTART:
+                    flag = DTLS_FLAGS_SERVER_LISTEN;
+                    dtls_context_clnt = NULL;
+                    DEBUG("DTLS Server listening!\n");
+                    msg_reply(&msg_rcvd,&msg_rcvd);
+                    xtimer_usleep(500U);
+                break;
+
+                case DTLS_MSG_CLIENT_HNDSHK: /* Nothing to do */
+#endif
                 default:
                     break;
             }
         }
 
+#ifndef MODULE_GNRC_DTLS
         _listen(&_sock);
+#else
+
+    if ( (flag == DTLS_FLAGS_SERVER_LISTEN) && !dtls_context_clnt ) {
```

Comment:
Once a CoAP command is given, the new thread will generate a new DTLS session or use a previous one for the client side, but also disable the DTLS server side of the sensor.  And I want some opinions for  this behavior. 

The logical would be that behavior was of a polling, by first listening the server sock and then the client sock. However, I fear that this could exhaust the sensor resources  as handling resources for both sides could be too much (and currently the risk is very high). 

Another idea I was thinking is to leave the polling (`if - if` instead of `if - else if`) with ANDs operators, but is still the same behavior.